native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/tasks/BuildNativeImageTask.java

```diff
@@ -189,7 +189,7 @@ private List<String> buildActualCommandLineArgs() {
                 // a mapped value before the task was called, when we are actually calling it...
                 getProviders().provider(() -> getOutputDirectory().getAsFile().get().getAbsolutePath()),
                 getClasspathJar(),
-                getUseArgFile()).asArguments();
+                getUseArgFile()).setMajorJDKVersion(majorJDKVersion).asArguments();
```

Comment:
@melix I'm not sure how to do this right. Adding another Provider means we are changing the API. Maybe you have a better suggestion.

Reply:
It is ok to change the API. As far as I know, the only consumer is the Micronaut plugin. We've already broken the contract several times, but it's fine, as long as figuring out what value to provide is easy.